bots/pr/src/test/java/org/openjdk/skara/bots/pr/CheckTests.java

```diff
@@ -203,63 +192,49 @@ void multipleReviews(TestInfo testInfo) throws IOException {
 
             // Let the status bot inspect the PR
             TestBotRunner.runPeriodicItems(checkBot);
-            assertFalse(authorPr.body().contains("Reviewers"));
+            assertFalse(authorPr.store().body().contains("Reviewers"));
 
             // Approve it
             var reviewerPr = reviewer.pullRequest(authorPr.id());
             reviewerPr.addReview(Review.Verdict.APPROVED, "Reviewers");
             TestBotRunner.runPeriodicItems(checkBot);
 
-            // Refresh the PR and check that it has been approved
-            authorPr = author.pullRequest(authorPr.id());
-            assertTrue(authorPr.body().contains("Reviewers"));
+            // Check that it has been approved
+            assertTrue(authorPr.store().body().contains("Reviewers"));
 
             // Update the file after approval
             editHash = CheckableRepository.appendAndCommit(localRepo, "Now I've gone and changed it");
             localRepo.push(editHash, author.url(), "edit", true);
 
-            // Make sure that the push registered
-            var lastHeadHash = authorPr.headHash();
-            var refreshCount = 0;
-            do {
-                authorPr = author.pullRequest(authorPr.id());
-                if (refreshCount++ > 100) {
-                    fail("The PR did not update after the new push");
-                }
-            } while (authorPr.headHash().equals(lastHeadHash));
-
             // Check that the review is flagged as stale
             TestBotRunner.runPeriodicItems(checkBot);
-            authorPr = author.pullRequest(authorPr.id());
```

Comment:
What was this line even supposed to accomplish? Some kind of reload?

Reply:
I'm happy to try to clarify. With the old `TestPullRequest` implementation, some data was part of a snapshot and some wasn't, so fetching the PR again caused a reload and was necessary to see some kinds of changes that a bot would make. This is even more true with the new implementation, but since we have an explicit object representing the server side state of the PR, I think it's better if test code uses that explicitly to inspect state.